docs/sources/flow/reference/components/prometheus.integration.redis.md

```diff
@@ -0,0 +1,104 @@
+# modifying this section to retain the spaces.
+#
+# Ideally, in the future, we can fix the overflow issue with css rather than
+# injecting special characters.
+
+title: prometheus.​integration.redis
+---
+
+# prometheus.integration.redis
+The `prometheus.integration.redis` component embeds
+[redis_exporter](https://github.com/oliver006/redis_exporter) for collecting metrics from a Redis database.
+
+## Usage
+
+```river
+prometheus.integration.redis "LABEL" {
+}
+```
+
+## Arguments
+The following arguments can be used to configure the exporter's behavior.
+Omitted fields take their default values.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`redis_addr`                  | `string`   | Address (host and port) of the Redis instance to connect to. This can be a local Redis instance (e.g., `localhost:6379`) or the address of a remote server. | | yes
+`redis_user`                  | `string`   | User name to use for authentication (Redis ACL for Redis 6.0 and newer).  | | no
+`redis_password`              | `string`   | Password of the Redis instance. | | no
+`redis_password_file`         | `string`   | Path of a file containing a password. If defined, it takes precedence over redis_password. | | no
+`redis_password_map_file`     | `string`   | Path of a file containing a JSON object which maps Redis URIs to passwords (e.g., `{"redis://localhost:6379": "sample_password"}`). | | no
+`namespace`                   | `string`   | Namespace for the metrics.  | `"redis"` | no
+`config_command`              | `string`   | What to use for the CONFIG command. | `"CONFIG"` | no
+`check_keys`                  | `string`   | Comma-separated list of key-patterns to export value and length/size, searched for with SCAN. | | no
+`check_key_groups`            | `string`   | Comma-separated list of Lua regular expressions (regex) for grouping keys. When unset, no key groups are made. | | no
+`check_key_groups_batch_size` | `int`      | Check key or key groups batch size hint for the underlying SCAN. The option name reflects key groups for backwards compatibility, but this applies to both key and key groups. | `10000` | no
+`max_distinct_key_groups`     | `int`      | The maximum number of distinct key groups with the most memory utilization to present as distinct metrics per database. The leftover key groups are aggregated in the 'overflow' bucket. | `100` | no
+`check_single_keys`           | `string`   | Comma-separated list of single keys to export value and length/size. | | no
+`check_streams`               | `string`   | Comma-separated list of stream-patterns to export info about streams, groups, and consumers to search for with SCAN. | | no
+`check_single_streams`        | `string`   | Comma-separated list of single streams to export info about streams, groups, and consumers. | | no
+`count_keys`                  | `string`   | Comma-separated list of individual keys to export counts for. | | no
```

Comment:
It's a little weird to me how many of these fields are a comma-separated list. Should we change them to use a `[]string` instead? 

It would introduce more differences between the Flow component and the existing integration, but it might be a better user experience which is harder to get wrong in the config file. 

WDYT @spartan0x117 @captncraig?

Reply:
It'd definitely be a better user experience, generally. It _could_ be a tiny bit more tedious to switch from using the exporter directly or using the agent in non-flow-mode -> using flow, as it would require converting these values for the new flow config (granted that could be done pretty quickly using a simple script).

Reply:
If we wanted to do that, I just pushed a commit that changes the flow config to accept `[]string`s and converts them to comma-separated strings in `Convert()`. I also tested a few with a minimal local example and verified the data being sent.

If we decide not to go that route I'll just have to revert that last commit before merging.